src/pattern/mod.rs

```diff
@@ -0,0 +1,6 @@
+//!
```

Comment:
it would be nice to have a short description here of what exactly a `pattern` is.